src/cube_neon.h

```diff
@@ -1,6 +1,423 @@
+	uint8x16_t p = vandq_u8(edge2, p_bits);
+	uint8x16_t piece1 = vqtbl1q_u8(edge1, p);
+
+	// Calculate the orientation through XOR
+	uint8x16_t orien = vandq_u8(veorq_u8(edge2, piece1), eo_bit);
+
+	// Combine the results
+	uint8x16_t ret = vorrq_u8(vandq_u8(piece1, p_bits), orien);
+
+	// Mask to clear the last 32 bits of the result
+	uint8x16_t mask_last_32 = vsetq_lane_u32(0, vreinterpretq_u32_u8(ret), 3);
+	ret = vreinterpretq_u8_u32(mask_last_32);
+
+	return ret;
+}
+_static_inline uint8x16_t compose_corners_slim(uint8x16_t corner1, uint8x16_t corner2)
+{
+	// Masks
+	uint8x16_t p_bits = vdupq_n_u8(_pbits);
+	uint8x16_t cobits = vdupq_n_u8(_cobits);
+	uint8x16_t cobits2 = vdupq_n_u8(_cobits2);
+	uint8x16_t twist_cw = vdupq_n_u8(_ctwist_cw);
+
+	// Find the index and permutation
+	uint8x16_t p = vandq_u8(corner2, p_bits);
+	uint8x16_t piece1 = vqtbl1q_u8(corner1, p);
+
+	// Calculate the orientation
+	uint8x16_t aux = vaddq_u8(vandq_u8(corner2, cobits), vandq_u8(piece1, cobits));
+	uint8x16_t auy = vshrq_n_u8(vaddq_u8(aux, twist_cw), 2);
+	uint8x16_t orien = vandq_u8(vaddq_u8(aux, auy), cobits2);
+
+	// Combine the results
+	uint8x16_t ret = vorrq_u8(vandq_u8(piece1, p_bits), orien);
+
+	// Mask to clear the last 64 bits of the result
+	uint8x16_t mask_last_64 = vsetq_lane_u64(0, vreinterpretq_u64_u8(ret), 1);
+	ret = vreinterpretq_u8_u64(mask_last_64);
+
+	return ret;
+}
+_static_inline cube_t compose(cube_t c1, cube_t c2)
+{
+	cube_t ret = {0};
+
+	ret.edge = compose_edges_slim(c1.edge, c2.edge);
+	ret.corner = compose_corners_slim(c1.corner, c2.corner);
+
+	return ret;
+}
+
+_static_inline cube_t inverse(cube_t cube)
+{
+	uint8_t i, piece, orien;
+	cube_t ret;
+
+	// Temp arrays to store the NEON vectors
+	uint8_t edges[16];
+	uint8_t corners[16];
+
+	// Copy the NEON vectors to the arrays
+	vst1q_u8(edges, cube.edge);
+	vst1q_u8(corners, cube.corner);
+
+	uint8_t edge_result[16] = {0};
+	uint8_t corner_result[16] = {0};
+
+	// Process the edges
+	for (i = 0; i < 12; i++)
+	{
+		piece = edges[i];
+		orien = piece & _eobit;
+		edge_result[piece & _pbits] = i | orien;
+	}
+
+	// Process the corners
+	for (i = 0; i < 8; i++)
+	{
+		piece = corners[i];
+		orien = ((piece << 1) | (piece >> 1)) & _cobits2;
+		corner_result[piece & _pbits] = i | orien;
+	}
+
+	// Copy the results back to the NEON vectors
+	ret.edge = vld1q_u8(edge_result);
+	ret.corner = vld1q_u8(corner_result);
+
+	return ret;
+}
+
+_static_inline int64_t coord_co(cube_t c)
+{
+	// Temp array to store the NEON vector
+	uint8_t mem[16];
+	vst1q_u8(mem, c.corner);
+
+	int i, p;
+	int64_t ret;
+
+	for (ret = 0, i = 0, p = 1; i < 7; i++, p *= 3)
+		ret += p * (mem[i] >> _coshift);
+
+	return ret;
+}
+
+_static_inline int64_t
+coord_csep(cube_t c)
+{
+	// Temp array to store the NEON vector
+	uint8_t mem[16];
+	vst1q_u8(mem, c.corner);
+
+	int64_t ret = 0;
+	int i, p;
+	for (ret = 0, i = 0, p = 1; i < 7; i++, p *= 2)
+		ret += p * ((mem[i] & _csepbit) >> 2);
```

Comment:
Anche qui e in coord_eo() forse si riesce a fare senza loop, ma servono le istruzioni giuste. Per AVX2 ho usato `_mm256_movemask_epi8()`.